modules/tinymce/src/themes/silver/main/ts/ui/general/Button.ts

```diff
@@ -126,7 +121,7 @@ const renderButtonSpec = (
   const components = [ icon.getOrThunk(() => GuiFactory.text(translatedText)) ];
 
   // The old default is based on the now-deprecated 'primary' property. `buttonType` takes precedence now.
-  const buttonType = spec.buttonType.getOr(!spec.primary && !spec.borderless ? 'secondary' : 'primary');
+  const buttonType = spec.buttonType.getOr(!Optionals.equals(spec.buttonType, Optional.some('primary')) && !spec.borderless ? 'secondary' : 'primary');
```

Comment:
Can remove the comment above now too. Also, having an `Optionals.equals` inside a `getOr` doesn't seem like the right way to do the check

Reply:
Luke is more than just correct, `spec.buttonType` will by definition not equal `primary` (because it's guaranteed to be `none` in the `getOr`).